pipeline/DESIGN.md

```diff
@@ -0,0 +1,190 @@
+# Pipeline Architecture
+
+The pipeline architecture allows for defining features over Ceramic data as a series of transformations over that data.
+What follows is a descriptions of each phase of the ceramic pipeline the features it enables and its dependencies.
+
+## Overview
+
+Data enters Ceramic via its API and is stored into the raw_events table.
+From there various transformations are applied producing various intermediate tables.
+Each table schema is considered public API and provides access to kind of query pattern against the data.
+
+```mermaid
+graph LR;
+    raw_events;
+    streams;
+    chain_proofs;
+    conclusion_events;
+    model_schemas;
+    event_states;
+    stream_heads;
+    stream_states;
+
+    raw_events --> conclusion_events;
+    streams --> conclusion_events;
+    chain_proofs --> conclusion_events;
+    conclusion_events --> event_states;
+    model_schemas --> event_states;
+    event_states --> stream_heads;
+    stream_heads --> stream_states;
+```
+
+## raw_events
+
+The raw_events table contains a row for each event in a stream and contains the raw CAR data of the event.
+
+### Features
+
+* Access to the raw event data that can be used to validate signatures
```

Comment:
I think we landed on not storing any events with invalid signatures in the current code. Is this focused on the future where we store them (maybe with a ttl or something) and move them to the next phase of the pipeline?

Reply:
Yeah I think we keep with that strategy of not storing invalid events. However this table gives consumers direct access to the raw data so that if they do not trust that ceramic-one validated signatures correctly they can do it themselves. Without this table its impossible for anyone to validate signatures themselves.